crates/notion-core/src/inventory/serial.rs

```diff
@@ -0,0 +1,108 @@
+use std::collections::{BTreeSet, HashSet};
+use std::iter::FromIterator;
+use std::marker::PhantomData;
+use std::path::Path;
+
+use fs::read_dir_eager;
+use notion_fail::{Fallible, ResultExt};
+use path;
+use super::{NodeCollection, YarnCollection};
+
+use regex::Regex;
+use semver::Version;
+
+/// Reads the contents of a directory and returns the set of all versions found
+/// in the directory's listing by matching filenames against the specified regex
+/// and parsing the `version` named capture as a semantic version.
+///
+/// The regex should contain the `version` named capture by using the Rust regex
+/// syntax `?P<version>`.
+fn versions_matching(dir: &Path, re: &Regex) -> Fallible<BTreeSet<Version>> {
+    Ok(read_dir_eager(dir)?
+        .filter(|(_, metadata)| {
+            metadata.is_file()
+        })
+        .filter_map(|(entry, _)| {
+            if let Some(file_name) = entry.path().file_name() {
+                if let Some(caps) = re.captures(&file_name.to_string_lossy()) {
+                    return Some(Version::parse(&caps["version"]).unknown());
+                }
+            }
+            None
+        })
+        .collect::<Fallible<BTreeSet<Version>>>()?)
+}
+
+impl NodeCollection {
+    pub(crate) fn load() -> Fallible<Self> {
+        let re = Regex::new(r"(?x)
+            node
+            -
+            v(?P<version>\d+\.\d+\.\d+) # Node version
+            -
+            (?P<os>[a-z]+)              # operating system
+            -
+            (?P<arch>[a-z0-9]+)         # architecture
+            \.(zip|tar\.gz)
```

Comment:
I was looking at download options to make sure this RE isn't missing anything, and it looks like there are `tar.xz` archives available for OSX and Linux, which are smaller download sizes. Might be an option for us to use in the future - I created #211 to track that.